streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -1616,4 +1623,25 @@ public <VOut> KStream<K, VOut> processValues(
             processNode,
             builder);
     }
+
+    @Override
+    public KStream<K, V> markAsPartitioned() {
+        final ProcessorParameters<? super K, ? super V, ?, ?> processorParameters =
+                new ProcessorParameters<>(new PassThrough<>(), PARTITION_PRESERVE_NAME + name);
```

Comment:
If we add `name` this could lead to conflict:
```
stream.markAsRepartition().filter();
stream.markAsRepartition().map();
```
This should be a valid program, however, both `filter` and `map` would generate the same processor name and thus we won't be able to compile it -- if we add the `Named` overload, we can use `NamedInternal(named).orElseGenerateWithPrefix` to generate a unique name like we do for other operator to avoid this issue.

Reply:
I will keep this open to remind myself to update the KIP after the review is complete.